src/app/lightning/index.ts

```diff
@@ -1,3 +1,18 @@
-export * from "./payment-status-checker"
-export * from "./lookup-invoice-by-hash"
-export * from "./lookup-payment-by-hash"
+import { wrapAsyncToRunInSpan, wrapToRunInSpan } from "@services/tracing"
+import * as LookupInvoiceByHash from "./lookup-invoice-by-hash"
+import * as LookupPaymentByHash from "./lookup-payment-by-hash"
+import { PaymentStatusChecker as PaymentStatusCheckerFn } from "./payment-status-checker"
+
+// creates an object with all the functions of imported modules
+const dynamicWrapper = { ...LookupInvoiceByHash, ...LookupPaymentByHash }
+
+for (const fn in dynamicWrapper) {
+  // wrap with async span, we need to be careful with functions that don't return a promise like PaymentStatusChecker
+  dynamicWrapper[fn] = wrapAsyncToRunInSpan({ fn: dynamicWrapper[fn] })
+}
+
+// Still we need to do a static export
+export const { PaymentStatusChecker, lookupInvoiceByHash, lookupPaymentByHash } = {
+  PaymentStatusChecker: wrapToRunInSpan({ fn: PaymentStatusCheckerFn }),
```

Comment:
I presume we're wrapping `PaymentStatusChecker` differently because it's running in sync? we could add the wrapper before the `export const` to normalize this, in a `allSyncFunctions` loop.

I guess this is just an example here and when agreed upon we would replicate for other apps?

Reply:
I did a different implementation, please check https://github.com/GaloyMoney/galoy/blob/13d967d7b89118d4648cc6c423e4084fca54e66a/src/app/index.ts